### ISSUE TITLE
🐛 Bug Report: Getting 500 when adding a new document thru the Appwrite UI

### ISSUE THREAD

**Issue by derek-diaz** [appwrite/appwrite#3044]
### 👟 Reproduction steps

 - I go to Database > select an existing collection
 - Under the documents tab I select "Add Document"
 - I type in the information and then click "Create"
 - I get a "Server Error" Message

### 👍 Expected behavior

I expected the document to be added to the collection/200. Or at least get an error message if anything is wrong with the input.

### 👎 Actual Behavior

When you click "Create" a "Server Error" message shows up with an error 500 (general_unknown) and if you quickly click on create again you get another error 500 but this time it says `AppwriteException: "message":"Error: There is already an active transaction","code":500`

# Error in Appwrite container
```
[Error] Timestamp: 2022-03-31T01:41:19+00:00
[Error] Method: POST
[Error] URL: /v1/database/collections/:collectionId/documents
[Error] Type: TypeError
[Error] Message: Exception::__construct(): Argument #1 ($message) must be of type string, null given
[Error] File: @swoole-src/library/core/Database/PDOStatementProxy.php
[Error] Line: 64
```

# Server Response
```json
{
  "message": "Server Error",
  "code": 500,
  "type": "general_unknown",
  "version": "0.13.4",
  "file": "@swoole-src/library/core/Database/PDOStatementProxy.php",
  "line": 64,
  "trace": [
    {
      "file": "@swoole-src/library/core/Database/PDOStatementProxy.php",
      "line": 64,
      "function": "__construct",
      "class": "Exception",
      "type": "->",
      "args": [
        null,
        null
      ]
    },
    {
      "file": "/usr/src/code/vendor/utopia-php/database/src/Database/Adapter/MariaDB.php",
      "line": 451,
      "function": "__call",
      "class": "Swoole\\Database\\PDOStatementProxy",
      "type": "->",
      "args": [
        "execute",
        []
      ]
    },
    {
      "file": "/usr/src/code/vendor/utopia-php/database/src/Database/Database.php",
      "line": 827,
      "function": "createDocument",
      "class": "Utopia\\Database\\Adapter\\MariaDB",
      "type": "->",
      "args": [
        "collection_1",
        {
          "pay-type": "per-year",
          "company-name": "some name",
          "base-pay": 10000,
          "job-title": "some title",
          "city": "some city",
          "state": "some state",
          "employment-type": "full-time",
          "is-still-employer": true,
          "company-rating": 1,
          "would-recommend": true,
          "single-word-description": "word",
          "$collection": "collection_1",
          "$id": "624506bf846bbe9d1d83",
          "$read": [],
          "$write": []
        }
      ]
    },
    {
      "file": "/usr/src/code/app/controllers/api/database.php",
      "line": 1648,
      "function": "createDocument",
      "class": "Utopia\\Database\\Database",
      "type": "->",
      "args": [
        {
          "name": "collection_1",
          "attributes": {
            "4": {
              "$id": "pay-type",
              "key": "pay-type",
              "type": "string",
              "size": 9,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "enum",
              "formatOptions": {
                "elements": [
                  "per-year",
                  "per-hour",
                  "per-month"
                ]
              },
              "filters": []
            },
            "5": {
              "$id": "company-name",
              "key": "company-name",
              "type": "string",
              "size": 255,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            },
            "6": {
              "$id": "base-pay",
              "key": "base-pay",
              "type": "double",
              "size": 0,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "floatRange",
              "formatOptions": {
                "min": -1.7976931348623157e+308,
                "max": 1.7976931348623157e+308
              },
              "filters": []
            },
            "7": {
              "$id": "job-title",
              "key": "job-title",
              "type": "string",
              "size": 255,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            },
            "8": {
              "$id": "city",
              "key": "city",
              "type": "string",
              "size": 255,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            },
            "9": {
              "$id": "state",
              "key": "state",
              "type": "string",
              "size": 255,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            },
            "10": {
              "$id": "employment-type",
              "key": "employment-type",
              "type": "string",
              "size": 9,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "enum",
              "formatOptions": {
                "elements": [
                  "full-time",
                  "part-time",
                  "temporary",
                  "intern",
                  "contract"
                ]
              },
              "filters": []
            },
            "11": {
              "$id": "is-still-employer",
              "key": "is-still-employer",
              "type": "boolean",
              "size": 0,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            },
            "12": {
              "$id": "company-rating",
              "key": "company-rating",
              "type": "integer",
              "size": 4,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "intRange",
              "formatOptions": {
                "min": 0,
                "max": 5
              },
              "filters": []
            },
            "13": {
              "$id": "would-recommend",
              "key": "would-recommend",
              "type": "boolean",
              "size": 0,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            },
            "14": {
              "$id": "single-word-description",
              "key": "single-word-description",
              "type": "string",
              "size": 255,
              "required": true,
              "default": null,
              "signed": true,
              "array": false,
              "format": "",
              "formatOptions": [],
              "filters": []
            }
          },
          "indexes": [],
          "$read": [
            "role:all"
          ],
          "$write": [
            "role:all"
          ],
          "$id": "collection_1",
          "$internalId": "23",
          "$collection": "_metadata"
        },
        {
          "pay-type": "per-year",
          "company-name": "some name",
          "base-pay": 10000,
          "job-title": "some title",
          "city": "some city",
          "state": "some state",
          "employment-type": "full-time",
          "is-still-employer": true,
          "company-rating": 1,
          "would-recommend": true,
          "single-word-description": "word",
          "$collection": "collection_1",
          "$id": "624506bf846bbe9d1d83",
          "$read": [],
          "$write": []
        }
      ]
    },
    {
      "file": "/usr/src/code/vendor/utopia-php/database/src/Database/Validator/Authorization.php",
      "line": 160,
      "function": "{closure}",
      "args": []
    },
    {
      "file": "/usr/src/code/app/controllers/api/database.php",
      "line": 1648,
      "function": "skip",
      "class": "Utopia\\Database\\Validator\\Authorization",
      "type": "::",
      "args": [
        {}
      ]
    },
    {
      "file": "/usr/src/code/vendor/utopia-php/framework/src/App.php",
      "line": 593,
      "function": "{closure}",
      "args": [
        "unique()",
        "initial-company-data",
        {
          "pay-type": "per-year",
          "company-name": "some name",
          "base-pay": 10000,
          "job-title": "some title",
          "city": "some city",
          "state": "some state",
          "employment-type": "full-time",
          "is-still-employer": true,
          "company-rating": 1,
          "would-recommend": true,
          "single-word-description": "word",
          "$collection": "initial-company-data",
          "$id": "624506bf846bbe9d1d83",
          "$read": [],
          "$write": []
        },
        [],
        [],
        {},
        {},
        {
          "name": "REDACTED",
          "email": "REDACTED",
          "status": true,
          "password": "REDACTED",
          "passwordUpdate": REDACTED,
          "prefs": [],
          "registration": REDACTED,
          "emailVerification": false,
          "reset": false,
          "sessions": [
            {
              "$id": "REDACTED",
              "userId": "REDACTED",
              "provider": "email",
              "providerUid": "REDACTED",
              "secret": "REDACTED",
              "expire": 1680220142,
              "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.84 Safari/537.36",
              "ip": "REDACTED",
              "countryCode": "us",
              "osCode": "WIN",
              "osName": "Windows",
              "osVersion": "10",
              "clientType": "browser",
              "clientCode": "CH",
              "clientName": "Chrome",
              "clientVersion": "99.0",
              "clientEngine": "Blink",
              "clientEngineVersion": "",
              "deviceName": "desktop",
              "deviceBrand": "",
              "deviceModel": "",
              "$read": [
                "user:REDACTED"
              ],
              "$write": [
                "user:REDACTED"
              ],
              "$collection": "sessions",
              "providerAccessToken": null,
              "providerAccessTokenExpiry": null,
              "providerRefreshToken": null
            },
            {
              "$id": "REDACTED",
              "userId": "REDACTED",
              "provider": "email",
              "providerUid": "REDACTED",
              "secret": "REDACTED",
              "expire": 1680220553,
              "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.84 Safari/537.36",
              "ip": "REDACTED",
              "countryCode": "us",
              "osCode": "WIN",
              "osName": "Windows",
              "osVersion": "10",
              "clientType": "browser",
              "clientCode": "CH",
              "clientName": "Chrome",
              "clientVersion": "99.0",
              "clientEngine": "Blink",
              "clientEngineVersion": "",
              "deviceName": "desktop",
              "deviceBrand": "",
              "deviceModel": "",
              "$read": [
                "user:REDACTED"
              ],
              "$write": [
                "user:REDACTED"
              ],
              "$collection": "sessions",
              "providerAccessToken": null,
              "providerAccessTokenExpiry": null,
              "providerRefreshToken": null
            }
          ],
          "tokens": [],
          "memberships": [
            {
              "$id": "REDACTED",
              "$read": [
                "user:REDACTED",
                "team:REDACTED"
              ],
              "$write": [
                "user:REDACTED",
                "team:REDACTED/owner"
              ],
              "userId": "REDACTED",
              "teamId": "REDACTED",
              "roles": [
                "owner"
              ],
              "invited": 1648684179,
              "joined": 1648684179,
              "confirm": true,
              "secret": "",
              "search": "REDACTED",
              "$collection": "memberships"
            }
          ],
          "search": "REDACTED",
          "deleted": false,
          "$read": [
            "role:all"
          ],
          "$write": [
            "user:6244ec6e451bb1dba941"
          ],
          "$id": "6244ec6e451bb1dba941",
          "$internalId": "1",
          "$collection": "users"
        },
        {},
        {},
        {},
        "admin"
      ]
    },
    {
      "file": "/usr/src/code/vendor/utopia-php/framework/src/App.php",
      "line": 694,
      "function": "execute",
      "class": "Utopia\\App",
      "type": "->",
      "args": [
        {},
        {}
      ]
    },
    {
      "file": "/usr/src/code/app/http.php",
      "line": 249,
      "function": "run",
      "class": "Utopia\\App",
      "type": "->",
      "args": [
        {},
        {}
      ]
    }
  ]
}
```

# Collection JSON
```json
{
    "$id": "initial-company-data",
    "$read": [
        "role:all"
    ],
    "$write": [
        "role:all"
    ],
    "name": "initial-company-data",
    "enabled": true,
    "permission": "collection",
    "attributes": [
        {
            "key": "pay-type",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "elements": [
                "per-year",
                "per-hour",
                "per-month"
            ],
            "format": "enum",
            "default": null
        },
        {
            "key": "company-name",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "size": 255,
            "default": null
        },
        {
            "key": "base-pay",
            "type": "double",
            "status": "available",
            "required": true,
            "array": false,
            "min": -1.7976931348623157e+308,
            "max": 1.7976931348623157e+308,
            "default": null
        },
        {
            "key": "job-title",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "size": 255,
            "default": null
        },
        {
            "key": "city",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "size": 255,
            "default": null
        },
        {
            "key": "state",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "size": 255,
            "default": null
        },
        {
            "key": "employment-type",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "elements": [
                "full-time",
                "part-time",
                "temporary",
                "intern",
                "contract"
            ],
            "format": "enum",
            "default": null
        },
        {
            "key": "is-still-employer",
            "type": "boolean",
            "status": "available",
            "required": true,
            "array": false,
            "default": null
        },
        {
            "key": "company-rating",
            "type": "integer",
            "status": "available",
            "required": true,
            "array": false,
            "min": 0,
            "max": 5,
            "default": null
        },
        {
            "key": "would-recommend",
            "type": "boolean",
            "status": "available",
            "required": true,
            "array": false,
            "default": null
        },
        {
            "key": "single-word-description",
            "type": "string",
            "status": "available",
            "required": true,
            "array": false,
            "size": 255,
            "default": null
        }
    ],
    "indexes": []
}
```

### 🎲 Appwrite version

Version 0.13.x

### 💻 Operating system

Linux

### 🧱 Your Environment

The environment is hosted in DigitalOcean using the pre-built application template. 

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by Meldiron** [comment]
I don't think I can reproduce this. Can you try to simplify your collection setup (attributes) to as simple as possible, while still getting the error?

`AppwriteException: "message":"Error: There is already an active transaction","code":500` might be fine if you click quickly on low-core machines.

An error that is concerning is `Argument #1 ($message) must be of type string, null given`, and Ill be more than happy to investigate it once I successfully replicate it.

**Comment by derek-diaz** [comment]
So not having a `-` on the collection name should fix it? Does this also apply to the collection field names? 

I'll see if I can create a new collection with and without to see if this is the root cause. 👍

**Comment by derek-diaz** [comment]
So I did some trial and error. 

Collection Id: some-collection
Attribute name: some-attribute 
❌ Error 500

Collection Id: someCollection
Attribute name: some-attribute 
❌ Error 500

Collection Id: some-collection
Attribute name: someAttribute 
✅ 200

Collection Id: someCollection
Attribute name: someAttribute 
✅ 200


So far it seems that the root cause is that having a `-` in the attribute name breaks the app. 🤔

**Comment by Meldiron** [comment]
The issue has been addressed at the library level. Once the new database library version is released, Appwrite will be upgraded to include this bug fix.

**Comment by stnguyen90** [comment]
So has this been released?

**Comment by stnguyen90** [comment]
@derek-diaz Are you still experiencing this problem after upgrading?

**Comment by lgecass** [comment]
In 0.15.2 this problem still occurs

**Comment by lgecass** [comment]
![image](https://user-images.githubusercontent.com/67883383/180656180-f80fff30-56df-4382-8310-b39806288a61.png)
here is the full Converation around this issue

**Comment by stnguyen90** [comment]
I can confirm the original issue is resolved as of 0.15.2:

<img width="1061" alt="image" src="https://user-images.githubusercontent.com/1477010/180865604-b9092646-7bd2-404d-bb4b-f40ab9dbb7df.png">

@lgecass, your issue was around querying so I created another issue for that.
